From 4c8ffcc996c9b1098c42401786cb1ec12d0552f1 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 10 Mar 2014 16:04:37 +0000
Subject: [PATCH 19/28] csys_fs: fix fd leaks in lseek error return path

Fix fd leaks in the lseek error return paths of
csys_fs::write() and csys_fs::read()

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_sys_fs.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/thd_sys_fs.cpp b/src/thd_sys_fs.cpp
index fb32dcb..89cf8fd 100644
--- a/src/thd_sys_fs.cpp
+++ b/src/thd_sys_fs.cpp
@@ -51,6 +51,7 @@ long long data) {
 	}
 	if (::lseek(fd, position, SEEK_CUR) == -1) {
 		thd_log_warn("sysfs write failed %s\n", path.c_str());
+		close(fd);
 		return -errno;
 	}
 	int ret = ::write(fd, &data, sizeof(data));
@@ -92,6 +93,7 @@ int csys_fs::read(const std::string &path, unsigned int position, char *buf,
 	}
 	if (::lseek(fd, position, SEEK_CUR) == -1) {
 		thd_log_warn("sysfs read failed %s\n", path.c_str());
+		close(fd);
 		return -errno;
 	}
 	int ret = ::read(fd, buf, len);
-- 
1.9.0

